sbi/inference/snpe/snpe_base.py

```diff
@@ -157,19 +157,28 @@ def __call__(
 
         for round_, num_sims in enumerate(num_sims_per_round):
 
-            # Run simulations for the round.
-            theta, x, prior_mask = self._run_simulations(round_, num_sims)
-            x_shape = x_shape_from_simulation(x)
+            # Run simulations for the round and prepend pre-simulated data.
+            theta, x = self._run_simulations(round_, num_sims)
+            theta, x = self._prepend_presimulated(theta, x)
+
+            # Check for NaNs in simulations.
+            is_valid_x, num_nans, num_infs = handle_invalid_x(x, exclude_invalid_x)
+            warn_on_invalid_x(num_nans, num_infs, exclude_invalid_x)
+
+            self._append_to_round_bank(theta, x, round_)
```

Comment:
Shouldn't only valid theta/x be appended? I'd move the part that checks for the validity inside `_append_to_round_bank`

Reply:
Good idea to move this!